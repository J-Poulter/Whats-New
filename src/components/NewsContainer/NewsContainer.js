import React from 'react';
import './NewsContainer.css'
import NewsArticle from '../NewsArticle/NewsArticle';

const NewsContainer = (props) => {

  return (
    <section> 
      {
        props.local.map(article => (
          <NewsArticle 
            id={article.id}
            img={article.img}
            headling={article.headline}
            description={article.description}
            url={article.url}
          />
        ))
      }
    </section>
  )
}

export default NewsContainer;