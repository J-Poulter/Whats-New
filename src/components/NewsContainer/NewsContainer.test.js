import React from 'react';
import ReactDOM from 'react-dom';
import NewsContainer from './NewsContainer';

import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('NewsContainer', () => {
  it('should render an element structure that matches the snapshot', () => {
    let mockCategories = [
      { title: 'entertainment', dataset: 'entertainment', iconImg: 'â˜…' },
      { title: 'local', dataset: 'local', iconImg: 'ðŸ’¡' }
    ];
    const snap = render(<NewsContainer 
      category={mockCategories}
    />)
    expect(snap).toMatchSnapshot();
  });

});
