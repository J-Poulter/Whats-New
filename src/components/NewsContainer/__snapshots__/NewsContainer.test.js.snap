// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer should render an element structure that matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <article
          class="articleCard"
        >
          <img
            alt="Failed to Load"
            class="articleImg"
          />
          <div
            class="cardInfo"
          >
            <h2
              class="articleTitle"
            />
            <p
              class="articleDescription"
            />
          </div>
          <div
            class="cardLink"
          >
            <a>
              Link To Article →
            </a>
          </div>
        </article>
        <article
          class="articleCard"
        >
          <img
            alt="Failed to Load"
            class="articleImg"
          />
          <div
            class="cardInfo"
          >
            <h2
              class="articleTitle"
            />
            <p
              class="articleDescription"
            />
          </div>
          <div
            class="cardLink"
          >
            <a>
              Link To Article →
            </a>
          </div>
        </article>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <article
        class="articleCard"
      >
        <img
          alt="Failed to Load"
          class="articleImg"
        />
        <div
          class="cardInfo"
        >
          <h2
            class="articleTitle"
          />
          <p
            class="articleDescription"
          />
        </div>
        <div
          class="cardLink"
        >
          <a>
            Link To Article →
          </a>
        </div>
      </article>
      <article
        class="articleCard"
      >
        <img
          alt="Failed to Load"
          class="articleImg"
        />
        <div
          class="cardInfo"
        >
          <h2
            class="articleTitle"
          />
          <p
            class="articleDescription"
          />
        </div>
        <div
          class="cardLink"
        >
          <a>
            Link To Article →
          </a>
        </div>
      </article>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
